src/maker/server.rs

```diff
@@ -338,52 +336,67 @@ fn setup_fidelity_bond(maker: &Arc<Maker>, maker_address: &str) -> Result<(), Ma
     Ok(())
 }
 
-/// Keep checking if the Bitcoin Core RPC connection is live. Sets the global `accepting_client` flag as per RPC connection status.
-///
-/// This will not block. Once Core RPC connection is live, accepting_client will set as `true` again.
-fn check_connection_with_core(
-    maker: Arc<Maker>,
-    accepting_clients: Arc<AtomicBool>,
-) -> Result<(), MakerError> {
-    let mut rpc_ping_success = false;
-    let mut i = 0;
+/// Checks if the maker has enough liquidity for swaps.
+/// If funds are below the minimum required, it repeatedly prompts the user to add more
+/// until the liquidity is sufficient.
+fn check_swap_liquidity(maker: &Maker) -> Result<(), MakerError> {
+    let sleep_incremental = 10;
+    let mut sleep_duration = 0;
+    let addr = maker.get_wallet().write()?.get_next_external_address()?;
     while !maker.shutdown.load(Relaxed) {
-        // If connection is disrupted keep trying at heart_beat_interval (3 sec).
-        // If connection is live, keep tring at rpc_ping_interval (60 sec).
-        let trigger_count = match rpc_ping_success {
-            true => RPC_PING_INTERVAL.as_secs() / HEART_BEAT_INTERVAL.as_secs(),
-            false => 1,
-        };
+        maker.get_wallet().write()?.sync_no_fail();
+        let offer_max_size = maker.get_wallet().read()?.store.offer_maxsize;
 
-        if i >= trigger_count || i == 0 {
-            if let Err(e) = maker.wallet.read()?.rpc.get_blockchain_info() {
-                log::error!(
-                    "[{}] RPC Connection failed. Reattempting {}",
-                    maker.config.network_port,
-                    e
-                );
-                rpc_ping_success = false;
-            } else {
-                if !rpc_ping_success {
-                    log::info!(
-                        "[{}] Bitcoin Core RPC connection is back online.",
-                        maker.config.network_port
-                    );
-                }
-                rpc_ping_success = true;
-            }
-            accepting_clients.store(rpc_ping_success, Relaxed);
-            i = 0;
+        let min_required = maker.config.min_swap_amount;
+        if offer_max_size < min_required {
+            log::warn!(
+                "Low Swap Liquidity | Min: {} sats | Available: {} sats. Add funds to {:?}",
+                min_required,
+                offer_max_size,
+                addr
+            );
+
+            sleep_duration = (sleep_duration + sleep_incremental).min(10 * 60); // Capped at 1 Block interval
```

Comment:
We don't need to wait for this long. The sleep be of a static 1 min interval, as the funds will be identified even if the tx is in mempool and that's all right. We can always fund coinswap from an unconfirmed utxo, as we will wait for the funding tx confirmation. That will confirm both the incoming fund tx and coinswap funding tx together. 

So we can iterate faster to capture in-mempool balance.

Reply:
Add a log here saying "next sync in 1 min", so user knows that its gonna sync again, and don't panic if the error doesn't go away immediately after spending funds.